addons/mrp/models/mrp_production.py

```diff
@@ -456,8 +468,15 @@ def create(self, values):
 
     @api.multi
     def unlink(self):
+        if any(production.state == 'done' for production in self):
+            raise UserError(_('Cannot delete a manufacturing order in done state.'))
+        self.action_cancel()
         if any(production.state != 'cancel' for production in self):
-            raise UserError(_('Cannot delete a manufacturing order not in cancel state'))
+            raise UserError(_('Same manufacturing orders cannot be deleted. Try to cancel them before.'))
```

Comment:
It would be better if you express MO that failed.
``` python3
productions_done = [p for p in self if p.state != 'cancel']
if productions_done:
    raise UserError(_('%s cannot be deleted. Try to cancel them before.') % (productions_done.display_name))
```